src/dash-table/derived/data/virtualized.ts

```diff
@@ -0,0 +1,63 @@
+import * as R from 'ramda';
+
+import { memoizeOneFactory } from 'core/memoizer';
+import {
+    IDerivedData,
+    IUserInterfaceViewport,
+    IUserInterfaceCell,
+    IVirtualizedDerivedData
+} from 'dash-table/components/Table/props';
+
+const getter = (
+    virtualization: boolean,
+    uiCell: IUserInterfaceCell | undefined,
+    uiHeaders: IUserInterfaceCell[] | undefined,
+    uiViewport: IUserInterfaceViewport | undefined,
+    viewport: IDerivedData
+): IVirtualizedDerivedData => {
+    if (!virtualization) {
+        return {
+            ...viewport,
+            offset: { rows: 0, columns: 0 },
+            padding: {
+                rows: { before: 0, after: 0 }
+            }
+        };
+    }
+
+    if (!uiViewport || !uiCell) {
+        return {
+            data: viewport.data.slice(0, 1),
+            indices: viewport.indices.slice(0, 1),
+            offset: { rows: 0, columns: 0 },
+            padding: {
+                rows: { before: 0, after: 0 }
+            }
+        };
+    }
+
+    const headersHeight = R.sum(R.map(h => h.height, uiHeaders || []));
+
+    const scrollTop = Math.max(uiViewport.scrollTop - headersHeight, 0);
+    const headersVisible = Math.max(headersHeight - uiViewport.scrollTop, 0);
+
+    let start = Math.floor(scrollTop / uiCell.height);
+    let end = Math.ceil(((uiViewport.height - headersVisible) + scrollTop) / uiCell.height);
```

Comment:
The headers provided are the ones present in the content section of the table -- if all header rows are accounted by `n_fixed_rows` they will not be here -- hence the lack of checks for n_fixed_rows, etc. the check happens implicitly before